### ISSUE TITLE
Arbitrary User Input?

### ISSUE THREAD

**Issue by 2br-2b** [pydot/pydot#357]
I'm curious - what's pydot's policy on data sanitization? Should I trust pydot to sanitize user input or should I do that myself?

Sanitizing data is always a good idea anyway, but I just wanted to check

**Comment by lkk7** [comment]
I think it's safe to assume for the user that we don't sanitize input at all, as pydot is just a wrapper.
Maybe it's a good idea to put this information in the README.

**Comment by 2br-2b** [comment]
That's a good assumption to know about. Maybe it should go in secuirty.md? I think that's a matter of preference, unless you put it somewhere near the bottom of the readme.

https://github.com/pydot/pydot/blob/d8645e72a3c97bdfd8a4abeaa7118090e1723e3c/src/pydot/core.py#L218-L226 was what I was most worried about, but per https://stackoverflow.com/a/52022409/10442089 , this should be safe from injection

**Comment by lkk7** [comment]
Oh, that's what you meant. If I remember correctly, there is no user input in that command, all parameters are "fixed".
But still, it's a good thing to look at and document.

**Comment by 2br-2b** [comment]
Yeah, my current use case is a adding this to my storywriting Discord bot. I'm planning on using this to let users create character graphs to keep track of what's going on. Given how insane the stories generally end up being (the bot chooses random server members and to continue writing one big story, so it bounces around a bunch), I think this will be a huge help for a lot of servers:

![image](https://github.com/pydot/pydot/assets/43968991/2347e3f2-1699-4d11-a1c8-e8681c40f1c9)

My concern is whether allowing users arbitrary access to create custom labels will open me up to command injection, or whether I should sanitize user input before passing it to Graphviz

This library has been doing a quite a bit of heavy lifting for that feature btw! You (along with the past maintainers and the Graphviz developers) have done a great job with this library!

**Comment by ferdnyc** [comment]
I think you should be "as safe as graphviz is" from the possibility of code injection, which should be pretty safe.

The `subprocess` call, as you noted, is safe (and as @lkk7 said, it includes no user input in the arguments).

And even graphviz's "HTML-like labels" (which we don't even parse other than as opaque string values, so there's no chance of us executing anything untoward) — are [referred to that way](https://graphviz.org/doc/info/shapes.html#html) specifically because they implement, "by hand" in the graphviz source, only a very small subset of HTML. Nobody's going to manage any script injections or anything like that, simply because it doesn't support any scripting at all, and will cheerfully ignore any tags other than the dozen or so it supports. The most advanced markup available is `<TABLE>`.
